ph5/utilities/fix_srm.py

```diff
@@ -0,0 +1,256 @@
+#
+# Lan Dam, November 2020
+#
+
+import argparse
+import os
+import sys
+import logging
+import tables
+
+from ph5.core import ph5api, experiment
+from ph5.utilities import nuke_table, kef2ph5 as K2T, tabletokef as T2K
+from ph5 import LOGGING_FORMAT
+
+PROG_VERSION = '2020.325'
+LOGGER = logging.getLogger(__name__)
+
+
+def get_args():
+    parser = argparse.ArgumentParser(
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.usage = ("fix_srm --nickname ph5-file-prefix [options]")
+
+    parser.description = ("Change sample rate multiplier=0 to 1 in"
+                          "Das table(s) or Array table(s). \nVersion: {0}"
+                          .format(PROG_VERSION))
+
+    parser.add_argument("-n", "--nickname", dest="ph5_file_prefix",
+                        help="The ph5 file prefix (experiment nickname).",
+                        metavar="ph5_file_prefix", required=True)
+
+    parser.add_argument("-p", "--path", dest="ph5_path",
+                        help="Path to ph5 files. Default to current "
+                             "directory.",
+                        metavar="ph5_path", default=".")
+
+    group = parser.add_mutually_exclusive_group(required=True)
+
+    group.add_argument("--all_das", dest='all_das', action='store_true',
+                       default=False,
+                       help=r"Fix sample_rate_multiplier_i in all Das tables"
+                            "\n/Experiment_g/Receivers_g/Das_g_xxx/Das_t.")
+
+    group.add_argument("-D", "--Das_t", dest="das_t_", metavar="das",
+                       help=r"Fix sample_rate_multiplier_i in Das table"
+                            "\n/Experiment_g/Receivers_g/Das_g_[das]/Das_t.")
+
+    group.add_argument("--all_arrays", dest='all_arrays', action='store_true',
+                       default=False,
+                       help=r"Fix sample_rate_multiplier_i in all Array "
+                            "table\n /Experiment_g/Sorts_g/Array_t_xxx .")
+
+    group.add_argument("-A", "--Array_t_", dest="array_t_", metavar="n",
+                       help=r"Fix sample_rate_multiplier_i in Array table"
+                            "\n /Experiment_g/Sorts_g/Array_t_[n].",
+                       type=int)
+
```

Comment:
I think that only the --all flag should be available. Metadata inconsistencies seems to be one of the biggest issues in PH5 and adding options to create more inconsistencies seems like it could lead to problems in the future. What do you think about this?

Reply:
In that case, we don't even need --all flag. Just run `fix_srm -n master.ph5 -p [path]`, it will fix the whole ph5 .

Reply:
Add a verification check so it is not accidentally run.